### PR TITLE
use an errgroup instead of a WaitGroup

### PR DESCRIPTION
**Pull request by kevinburke**
An errgroup makes it easy to propagate errors when any individual download
fails, instead of printing them to the screen.

### CODE REVIEW

**Comment by ianmdawson** [comment]
looks good to me ✨  -- cc @benbuckman

File: circle/main.go [hunk 1]
```diff
@@ -91,21 +91,22 @@ func doDownload(flags *flag.FlagSet) error {
...
 	}
 	for _, art := range arts {
-		wg.Add(1)
-		go func(a *circle.CircleArtifact) {
-			if err := circle.DownloadArtifact(a, tempDir, remote.Path); err != nil {
-				fmt.Fprintln(os.Stderr, err)
-			}
-			wg.Done()
-		}(art)
+		art := art
```
Comment (benbuckman) [inline circle/main.go:100]: What does this do (assignment to self)?

**Review by benbuckman — APPROVED** [review]
One question, otherwise LGTM and thanks!

File: circle/main.go (hunk 1 above)
Comment (benbuckman) [reply circle/main.go:100]: @kevinburke ^
thanks

Comment (benbuckman) [reply circle/main.go:100]: (A link to language specs explaining this syntax would suffice)

Comment (kevinburke) [reply circle/main.go:100]: https://golang.org/doc/faq#closures_and_goroutines

Comment (benbuckman) [reply circle/main.go:100]: Huh, TIL, thanks!

**Review by benbuckman — APPROVED** [review]

**Pull request merged**